src/query_planner/src/lib.rs

```diff
@@ -141,3 +141,253 @@ impl AsRef<(Id, Id)> for TableId {
 
 #[cfg(test)]
 mod tests;
+
```

Comment:
I don't know if I return to this code. Currently, I am working on making `insert`s and `update`s more consistent.
For example: right now in master `insert into table (col1, col2) values (1, 2);` column index resolution is happening in `InsertCommand` directly, however, `update table set col1 = col1 * 2` column index resolution is happening in `ExpressionEvaluation`
For now I want to do the following: column info should be gathered in the `query_planner` so that 
`TableInserts` and `TableUpdates` structs from `plan.rs` have `column_indecies: Vec<(usize, ColumnDefinition)>` fields that would help resolve column indexes.
So this `Mapper` could be split into multiple parts.
`ScalarOp::transform` will transform `Expr::BinaryOp` into `ScalarOp`
`ScalarValue::transform` will transform `Expr::Value`, `Expr::Cast` and `Expr::UnaryOp` into `ScalarValue` and `InsertPlanner` and `UpdatePlanner` will take care of how to transform `SetExpr::Values(Vec<Value>)` and `Vec<Assingment>` into `Vec<(usize, ColumnDefinition)>` respectively.

Hope this make sense.